### PR TITLE
[Auth] Add a really ugly form to allow basic registration. Other misc framework changes.

### PR DESCRIPTION
**Pull request by cdeveas**
Something seems to be up with styling. I was trying to enforce codebase-wide styling with `husky`, which runs a git-hook and executes Prettier, automatically styling the codebase, but it was the wrong version. Did we align on tabs over spaces for this codebase? I'll do a separate PR with styling fixes, but when this is merged you will start to see `pretty-quick` being run after you commit. It'll automatically apply our Prettier styling to any files you changed.

### CODE REVIEW

File: bin/deploy.js [hunk 1]
```diff
@@ -138,11 +138,18 @@ async function getInvalidationStatus(invalidationId, distribution) {
  */
 function _deployOptions(env) {
 	const conf = _loadConfigForNodeEnv(env);
-	return {
+
+	const deployOptions = {
```
Comment (cdeveas) [inline bin/deploy.js]: I'm definitely a fan of spaces over tabs, for the record...

File: src/modules/auth/auth.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+		default:
+			return state;
+	}
+}
+
+// Action Creators
+
+/**
+ * @param {RegisterDTO} registerDto
```
Comment (cdeveas) [inline src/modules/auth/auth.js]: Stands for data transfer object. Basically represents a payload sent in a REST request.

File: src/modules/common/App.jsx [hunk 3]
```diff
@@ -4,50 +4,56 @@ import {ConnectedRouter} from 'react-router-redux';
...
 import HomeContainer from '../home/HomeContainer';
+import HealthContainer from '../health/HealthContainer';
+import RegisterContainer from '../register/RegisterContainer';
 
 // Import common styles
 import './styles/base.scss';
 
 export default function App({history}) {
-
-  const Nav = styled.nav`
```
Comment (cdeveas) [inline src/modules/common/App.jsx]: I didn't change this manually.. this was from Prettier styling based on our .prettierrc config. See my overall comment about styling; I'll make another PR which auto-formats the entire codebase after this is merged.

File: bin/deploy.js (hunk 1 above)
Comment (osebo) [reply bin/deploy.js]: Not to keep harping on this but I thought we said tabs... Don't mind spaces though.

Comment (cdeveas) [reply bin/deploy.js]: Nah I'll stick with tabs. Just triple checking. Perhaps I will come around to it.

**Pull request closed without merging**